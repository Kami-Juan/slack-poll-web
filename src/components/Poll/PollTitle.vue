<template>
  <div id="poll-title">
    <b-row align-h="center">
      <b-col cols=6>
        <input
          id="title-input"
          type="text"
          placeholder="Here goes your poll title"
          autocomplete="off"
          :value="value"
          @input="$emit('input', $event.target.value)"
          @keyup.enter="$emit('keyupEnter')"
        >
      </b-col>
    </b-row>
  </div>
</template>

<script>
export default {
  name: 'PollTitle',
  props: ['value']
}
</script>

<style>
#title-input {
  width: 100%;
  outline: none;
  border: 0;
  font-size: 48px;
}

#title-input {
  outline: none;
}
</style>
