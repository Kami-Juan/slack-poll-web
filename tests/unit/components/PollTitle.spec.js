import BootstrapVue from 'bootstrap-vue'
import { LayoutPlugin, ModalPlugin } from 'bootstrap-vue'

import { mount, createLocalVue } from '@vue/test-utils'
import PollTitle from './../../../src/components/Poll/PollTitle'

describe('test PollTitle.vue', () => {
  const localVue = createLocalVue()

  localVue.use(BootstrapVue)
  localVue.use(LayoutPlugin)
  localVue.use(ModalPlugin)

  it('should be render', async () => {
    const cmp = mount(PollTitle, {localVue})

    await cmp.vm.$nextTick()

    expect(cmp.element).toMatchSnapshot()
  })

  it('the poll title has the correct value', () => {
    const cmp = mount(PollTitle, {localVue, propsData: {
      value: 'I"m title'
    }})

    let titleInput = cmp.find('#title-input').element.value
    expect(titleInput).toEqual('I"m title')
  })

  it('set data in the poll title and fire the input event', () => {
    const cmp = mount(PollTitle, {localVue})

    const titleInput = cmp.find('#title-input')
    titleInput.setValue('I"m title two')

    expect(cmp.emitted().input).toBeTruthy()
    expect(cmp.emitted().input.length).toBe(1)
    expect(cmp.emitted().input[0]).toEqual(['I"m title two'])
  })

  it('set attribute in the component and the poll title get it', () => {
    const cmp = mount(PollTitle, {localVue, attrs: {
      type: 'password',
      name: 'title-input',
      placeholder: 'new placeholder'
    }})

    const titleInput = cmp.find('#title-input').attributes()

    expect(titleInput.type).not.toBe('password')
    expect(titleInput.name).toBe('title-input')
    expect(titleInput.placeholder).not.toBe('new placeholder')
  })

  it('set listener and in the component and the title input fire it', async () => {
    const clickHandler = jest.fn()

    const cmp = mount(PollTitle, {localVue, listeners: {
      keyup: clickHandler
    }})

    const pollInput = cmp.find('#title-input')
    pollInput.trigger('keyup.enter')

    expect(clickHandler).toHaveBeenCalled()
  })
})