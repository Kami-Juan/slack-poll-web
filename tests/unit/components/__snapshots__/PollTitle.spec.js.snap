// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PollTitle.vue should be render 1`] = `
<div
  id="poll-title"
>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-md-6 col-12"
    >
      <input
        autocomplete="off"
        id="title-input"
        placeholder="Here goes your poll title"
        type="text"
      />
    </div>
  </div>
</div>
`;
